### PR TITLE
Improve file explorer

### PR DESCRIPTION
**Pull request by OlshaMB**
I want to improve file explorer usability
# Roadmap

- [x] Create folder
- [x] Actions on file
    - [x] Create new file on folder
    - [x] Create new folder on folder
    - [x] Delete File

### CODE REVIEW

**Comment by OlshaMB** [comment]
@Cubxity what icons do you use for typstudio?

**Comment by Cubxity** [comment]
Hi, thank you for contributing! Please mark the PR as draft if this is still W.I.P. Also, `typstudio` uses `pnpm`, please remove `yarn.lock` from the PR.

**Comment by Cubxity** [comment]
> @Cubxity what icons do you use for typstudio?

Material Icons.

**Comment by OlshaMB** [comment]
> > @Cubxity what icons do you use for typstudio?
> 
> Material Icons.

Sidequestion: shouldn't you credit them?

**Comment by OlshaMB** [comment]
> Hi, thank you for contributing! Please mark the PR as draft if this is still W.I.P. Also, `typstudio` uses `pnpm`, please remove `yarn.lock` from the PR.

Could i add `yarn.lock` to `.gitignore`

**Comment by OlshaMB** [comment]
I think the pr is done. I also needed to confirm dialog to ask before deletion.

**Comment by Cubxity** [comment]
Hey, can you rebase and resolve the conflicts?

File: src/components/ShellModal.svelte [hunk 1]
```diff
@@ -45,8 +56,16 @@
...
           on:keyup={handleInputKeyUp}
+          bind:value={$text}
           autofocus
-        >
+        />
+      {:else if modal.type === "confirm"}
+        <div class="flex justify-between mt-3">
+          <button on:click={handleClose} class="rounded-md border-2 border-neutral-700 px-2 py-1"
+            >Cancel</button
+          >
```
Comment (caoimhebyrne) [inline src/components/ShellModal.svelte:66]: Formatting is awkward here:
```svelte
<button ...>
    Cancel
</button>
```

**Review by caoimhebyrne — CHANGES_REQUESTED** [review]
I have just a few nitpicks

File: src/components/ExplorerNode.svelte [hunk 2]
```diff
@@ -4,26 +4,77 @@
...
+  const handleClick = (e: MouseEvent) => {
+    e.stopPropagation();
     if (type === "directory") {
       expanded = !expanded;
     } else {
       shell.selectFile(path);
     }
   };
-
+  $: handleCreateFile = (e: MouseEvent) => {
```
Comment (Cubxity) [inline src/components/ExplorerNode.svelte:37]: Why do these functions use subscribe (`$`)?

**Review by Cubxity — CHANGES_REQUESTED** [review]
I believe that creating directories and files should be hidden in a context menu. This removes clutter from the UI. Same goes for deletion.

**Comment by OlshaMB** [comment]
> I believe that creating directories and files should be hidden in a context menu. This removes clutter from the UI. Same goes for deletion.

Maybe show the icons only on hover?

**Comment by Cubxity** [comment]
> > I believe that creating directories and files should be hidden in a context menu. This removes clutter from the UI. Same goes for deletion.
> 
> @Cubxity Maybe show the icons only on hover?
> This also will remove clutter, but unlike a context menu you would feel that you are creating it on a directory not on a file explorer

That sounds alright.

**Comment by Csega** [comment]
Hi Guys!

What is the status of this PR? Can you comment on it? Is it still in progress or just hanging?

Cheers,
Csega

**Comment by caoimhebyrne** [comment]
Are you continuing this PR @OlshaMB?

**Comment by OlshaMB** [comment]
yes

**Comment by OlshaMB** [comment]
Done

**Comment by OlshaMB** [comment]
@caoimhebyrne Done

**Comment by Cubxity** [comment]
Could you rebase/resolve conflicts?

**Comment by caoimhebyrne** [comment]
@OlshaMB are you planning on continuing to maintain this PR? If not, please let us know...

**Comment by OlshaMB** [comment]
> @OlshaMB are you planning on continuing to maintain this PR? If not, please let us know ☺️

No, I am not planning, I am not interested in typst right now(waiting for html output) and there are too much changes to merge

**Comment by caoimhebyrne** [comment]
Ok, I will close the PR and open an issue for improving the file explorer.

**Pull request closed without merging**